# Ignasius Maret Ismoyo

# -----------------------------------------------
# Import Library

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
%matplotlib inline
import warnings
warnings.filterwarnings("ignore")
import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

/kaggle/input/best-country-to-live-in-2024/best-countries-to-live-in-2024.csv


# -----------------------------------------------
# Upload Data

In [2]:
data = pd.read_csv('/kaggle/input/best-country-to-live-in-2024/best-countries-to-live-in-2024.csv')
data

population_2024  population_growthRate  land_area        country  \
0         1441719852                0.00916    3287590          India   
1         1425178782               -0.00035    9706961          China   
2          341814420                0.00535    9372610  United States   
3          279798049                0.00816    1904569      Indonesia   
4          245209815                0.01964     881912       Pakistan   
..               ...                    ...        ...            ...   
136           867605                0.01823       1862        Comoros   
137           661594                0.01043       2586     Luxembourg   
138           626102               -0.00061      13812     Montenegro   
139           536740                0.00313        316          Malta   
140           377689                0.00632     103000        Iceland   

            region  unMember  population_density  population_densityMi  \
0             Asia      True            484.9067             1255.9084   
1             Asia      True            151.2174              391.6530   
2    North America      True             37.3673               96.7813   
3             Asia      True            149.0254              385.9758   
4             Asia      True            318.0908              823.8551   
..             ...       ...                 ...                   ...   
136         Africa      True            466.2037             1207.4675   
137         Europe      True            256.9796              665.5772   
138         Europe      True             46.5503              120.5654   
139         Europe      True           1677.3125             4344.2394   
140         Europe      True              3.7458                9.7016   

                                         share_borders  Hdi2021  Hdi2020  \
0               AFG, BGD, BTN, MMR, CHN, NPL, PAK, LKA    0.633    0.642   
1    AFG, BTN, MMR, HKG, IND, KAZ, PRK, KGZ, LAO, M...    0.768    0.764   
2                                             CAN, MEX    0.921    0.920   
3                                        TLS, MYS, PNG    0.705    0.709   
4                                   AFG, CHN, IND, IRN    0.544    0.543   
..                                                 ...      ...      ...   
136                                                NaN    0.558    0.562   
137                                      BEL, FRA, DEU    0.930    0.924   
138                            ALB, BIH, HRV, KOS, SRB    0.832    0.826   
139                                                NaN    0.918    0.911   
140                                                NaN    0.959    0.957   

     WorldHappiness2022  
0                 3.777  
1                 5.585  
2                 6.977  
3                 5.240  
4                 4.516  
..                  ...  
136               4.609  
137               7.404  
138               5.547  
139               6.447  
140               7.557  

[141 rows x 12 columns]

# -----------------------------------------------
# Data Exploration

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 141 entries, 0 to 140
Data columns (total 12 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   population_2024        141 non-null    int64  
 1   population_growthRate  141 non-null    float64
 2   land_area              141 non-null    int64  
 3   country                141 non-null    object 
 4   region                 141 non-null    object 
 5   unMember               141 non-null    bool   
 6   population_density     141 non-null    float64
 7   population_densityMi   141 non-null    float64
 8   share_borders          129 non-null    object 
 9   Hdi2021                141 non-null    float64
 10  Hdi2020                141 non-null    float64
 11  WorldHappiness2022     141 non-null    float64
dtypes: bool(1), float64(6), int64(2), object(3)
memory usage: 12.4+ KB


In [4]:
print(f'Total Null data = {data.isnull().sum().sum()} cells')
print(f'Total NaN data = {data.isna().sum().sum()} cells')

Total Null data = 12 cells
Total NaN data = 12 cells


In [5]:
data.isnull().sum()

population_2024           0
population_growthRate     0
land_area                 0
country                   0
region                    0
unMember                  0
population_density        0
population_densityMi      0
share_borders            12
Hdi2021                   0
Hdi2020                   0
WorldHappiness2022        0
dtype: int64

In [6]:
data = data.drop(['share_borders'], axis=1)
data.isnull().sum()

population_2024          0
population_growthRate    0
land_area                0
country                  0
region                   0
unMember                 0
population_density       0
population_densityMi     0
Hdi2021                  0
Hdi2020                  0
WorldHappiness2022       0
dtype: int64

In [7]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 141 entries, 0 to 140
Data columns (total 11 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   population_2024        141 non-null    int64  
 1   population_growthRate  141 non-null    float64
 2   land_area              141 non-null    int64  
 3   country                141 non-null    object 
 4   region                 141 non-null    object 
 5   unMember               141 non-null    bool   
 6   population_density     141 non-null    float64
 7   population_densityMi   141 non-null    float64
 8   Hdi2021                141 non-null    float64
 9   Hdi2020                141 non-null    float64
 10  WorldHappiness2022     141 non-null    float64
dtypes: bool(1), float64(6), int64(2), object(2)
memory usage: 11.3+ KB


# -----------------------------------------------
# Estimate Total and Average World Population in 2024

In [8]:
data.sample(5)

population_2024  population_growthRate  land_area    country  \
69          12564689                0.00855     163610    Tunisia   
29          47473373               -0.00097     505992      Spain   
114          3717425               -0.00291      69700    Georgia   
134          1268467                0.00661       9251     Cyprus   
32          46057866                0.00620    2780400  Argentina   

            region  unMember  population_density  population_densityMi  \
69          Africa      True             80.8747              209.4654   
29          Europe      True             95.0310              246.1303   
114           Asia      True             53.4958              138.5542   
134         Europe      True            137.2800              355.5551   
32   South America      True             16.8298               43.5891   

     Hdi2021  Hdi2020  WorldHappiness2022  
69     0.731    0.737               4.516  
29     0.905    0.899               6.476  
114    0.802    0.802               4.973  
134    0.896    0.894               6.221  
32     0.842    0.840               5.967

In [9]:
total_wp2024 = np.round(data['population_2024'].sum() / 1000000000, 2)
print(f"Total World Population in 2024 is {total_wp2024} billion people's")

Total World Population in 2024 is 7.65 billion people's


In [10]:
avg_wp2024 = np.around(data['population_2024'].mean() / 1000000, 2)
print(f"Average of World Population in 2024 is {avg_wp2024} million people's")

Average of World Population in 2024 is 54.22 million people's


In [11]:
total_la = np.round(data['land_area'].sum() / 1000000, 2)
print(f"Total Land Area is {total_la} million km^2")

Total Land Area is 123.15 million km^2


In [12]:
avg_wp2024perla = np.around(data['population_2024'].sum() / data['land_area'].sum(), 1)
print(f"Density of World Population per km^2 in 2024 is {avg_wp2024perla} people")

Density of World Population per km^2 in 2024 is 62.1 people


In [13]:
fig_happiness_map = px.choropleth(data, locations='country', locationmode='country names',
                                  color='population_2024',
                                  title='Total Population in 2024',
                                  color_continuous_scale='Viridis',
                                  labels={'population_densityMi': 'Population Density'})

# Display the map
fig_happiness_map.show()

# -----------------------------------------------
# HDI 2020 & 2021, Pop Growth Rate and World Happiness

In [14]:
hdi2020 = np.around(data.Hdi2020.mean(), 4)
hdi2021 = np.around(data.Hdi2021.mean(), 4)

print(f'Average of HDI 2020 = {hdi2020}')
print(f'Average of HDI 2021 = {hdi2021}')

Average of HDI 2020 = 0.7376
Average of HDI 2021 = 0.7372


In [15]:
pop_gr = np.around(data.population_growthRate.mean(), 4)
print(f'Average of Population growth rate = {pop_gr}')

Average of Population growth rate = 0.0093


In [16]:
wh2022 = np.around(data.WorldHappiness2022.mean(), 2)
print(f'Average of World Happiness in 2022 = {wh2022} point')

Average of World Happiness in 2022 = 5.54 point


In [17]:
fig_happiness_map = px.choropleth(data, locations='country', locationmode='country names',
                                  color='Hdi2020',
                                  title='Human Development Index 2020',
                                  color_continuous_scale='Viridis',
                                  labels={'Hdi2020': 'HDI 2020'})

# Display the map
fig_happiness_map.show()

In [18]:
fig_happiness_map = px.choropleth(data, locations='country', locationmode='country names',
                                  color='Hdi2021',
                                  title='Human Development Index 2021',
                                  color_continuous_scale='Viridis',
                                  labels={'Hdi2021': 'HDI 2021'})

# Display the map
fig_happiness_map.show()

In [19]:
region_g_r = data.groupby(['region','unMember'])['WorldHappiness2022'].apply(lambda x: x.value_counts().idxmax()).sort_values(ascending=False).reset_index()
region_g_r

region  unMember  WorldHappiness2022
0        Oceania      True               7.162
1  North America      True               6.977
2  South America      True               6.293
3           Asia     False               5.425
4           Asia      True               5.377
5         Europe      True               5.199
6         Africa      True               5.048

* #### Oceania is the most happiness region in 2022 with World Happiness score is 7.162 point

In [20]:
country_g_r = data.groupby(['country','unMember'])['WorldHappiness2022'].apply(lambda x: x.value_counts().idxmax()).sort_values(ascending=False).reset_index().head(20)
country_g_r

country  unMember  WorldHappiness2022
0          Finland      True               7.821
1          Denmark      True               7.636
2          Iceland      True               7.557
3      Switzerland      True               7.512
4      Netherlands      True               7.415
5       Luxembourg      True               7.404
6           Sweden      True               7.384
7           Norway      True               7.365
8           Israel      True               7.364
9      New Zealand      True               7.200
10         Austria      True               7.163
11       Australia      True               7.162
12         Ireland      True               7.041
13         Germany      True               7.034
14          Canada      True               7.025
15   United States      True               6.977
16  United Kingdom      True               6.943
17  Czech Republic      True               6.920
18         Belgium      True               6.805
19          France      True               6.687

* #### Mostly countries in the region of Europe is on the list

In [21]:
fig_happiness_map = px.choropleth(data, locations='country', locationmode='country names',
                                  color='WorldHappiness2022',
                                  title='World Happiness Index 2022',
                                  color_continuous_scale='Viridis',
                                  labels={'WorldHappiness2022': 'World Happiness Index'})

# Display the map
fig_happiness_map.show()

In [22]:
tail_country_gr = (
    data.groupby("country")["population_growthRate"]
    .mean()
    .sort_values(ascending=False)
    .tail(5)
)
tail_country_gr

country
Hungary    -0.01588
Slovakia   -0.01594
Poland     -0.01961
Lebanon    -0.02519
Moldova    -0.03087
Name: population_growthRate, dtype: float64

* #### Moldova is the lowest population growth rate with -0.03087 point

In [23]:
head_country_gr = (
    data.groupby("country")["population_growthRate"]
    .mean()
    .sort_values(ascending=False)
    .head(5)
)
head_country_gr

country
Niger       0.03809
Ukraine     0.03247
Chad        0.03111
Mali        0.03100
Tanzania    0.02938
Name: population_growthRate, dtype: float64

* #### Niger is the highest population growth rate with 0.03809 point

In [24]:
fig_happiness_map = px.choropleth(data, locations='country', locationmode='country names',
                                  color='population_growthRate',
                                  title='Population Growth Rate',
                                  color_continuous_scale='Viridis',
                                  labels={'population_growthRate': 'Population Growth Rate'})

# Display the map
fig_happiness_map.show()

# -----------------------------------------------
# Total UN Member countries in every Region

In [25]:
data.sample(5)

population_2024  population_growthRate  land_area       country  region  \
96           6598071                0.01258     488100  Turkmenistan    Asia   
133          1301978                0.00109       2040     Mauritius  Africa   
31          46278751                0.01474    2381741       Algeria  Africa   
1         1425178782               -0.00035    9706961         China    Asia   
38          37473929                0.01426    2149690  Saudi Arabia    Asia   

     unMember  population_density  population_densityMi  Hdi2021  Hdi2020  \
96       True             14.0405               36.3650    0.745    0.741   
133      True            641.3685             1661.1443    0.802    0.804   
31       True             19.4306               50.3254    0.745    0.736   
1        True            151.2174              391.6530    0.768    0.764   
38       True             17.4322               45.1495    0.875    0.870   

     WorldHappiness2022  
96                5.474  
133               6.071  
31                5.122  
1                 5.585  
38                6.523

In [26]:
unMember = data.groupby('region')['unMember'].apply(lambda x: x.count()).sort_values(ascending=False).reset_index()
unMember

region  unMember
0         Africa        40
1         Europe        40
2           Asia        39
3  North America        11
4  South America         9
5        Oceania         2

In [27]:
fig_happiness_map = px.choropleth(data, locations='country', locationmode='country names',
                                  color='unMember',
                                  title='United Nations member',
                                  color_continuous_scale='Viridis',
                                  labels={'population_densityMi': 'World Happiness Index'})

# Display the map
fig_happiness_map.show()